service/chain_service.go

```diff
@@ -186,7 +186,7 @@ func (cs *chainService) createHeader(hash *domains.BlockHash, bs *domains.BlockH
 
 func (cs *chainService) previousHeader(bs *domains.BlockHeaderSource) (*domains.BlockHeader, error) {
 	h, err := cs.Repositories.Headers.GetHeaderByHash(bs.PrevBlock.String())
-	if h == nil && err != nil && err.Error() == "could not find hash" {
+	if h == nil && err != nil && err.Error() == "Header not found" {
```

Comment:
Can't it be done using `errors.Is` ?

I'm not a fun of comparing error messages especially when it is a part of some logic/algorithm.

Reply:
It can, sure let me change it out